xwiki-platform-core/xwiki-platform-rendering/xwiki-platform-rendering-macros/xwiki-platform-rendering-macro-include/src/main/java/org/xwiki/rendering/internal/macro/include/IncludeMacro.java

```diff
@@ -202,7 +202,23 @@ public void setDocumentDisplayer(DocumentDisplayer documentDisplayer)
                 references.pop();
             }
         }
-
+        
+        // Exclude First Heading
+        if (parameters.excludeFirstHeading()) {
+            if (result.getChildren().size() > 0) {
+                Block firstBlock = result.getChildren().get(0);
+                if (firstBlock instanceof SectionBlock && firstBlock.getChildren().size() > 0) {
```

Comment:
You can combine a lot of the "if"s. I doubt that this code will pass the checkstyle checks. Does it?

Reply:
Yes On combining if's there is no checkstyle voilation. I build using `mvn install -Pquality` and it builds successfully